lms/djangoapps/entitlements/models.py

```diff
@@ -0,0 +1,51 @@
+from django.contrib.auth.models import User
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from jsonfield import JSONField
+
+from .registry import registry
+from .scope import ScopeFactory
+
+
+class EntitlementModel(models.Model):
+    type = models.CharField(
+        max_length=255, help_text=_("Entitlement type")
+    )
+    scope_id = models.CharField(max_length=4000, help_text=_("ID of an object Entitlement is scoped to"))
+    parameters = JSONField(blank=True, default={})
+
+    class Meta(object):
+        app_label = "entitlements"
+
+    def get_entitlement(self):
+        entitlement_class = registry.get_entitlement_and_scope(self.type)
+        scope_strategy = ScopeFactory.make_scope_strategy(entitlement_class.SCOPE_TYPE)
+        return entitlement_class(self.scope_id, scope_strategy, **self.parameters)
+
+    def __unicode__(self):
+        return u"Entitlement type: {type}, scope_id: {scope_id}".format(type=self.type, scope_id=self.scope_id)
+
+
+class EntitlementGroup(models.Model):
+    ENTERPRISE_CUSTOMER = "enterprise_customer"
```

Comment:
So this is where I come back to the idea of using the Organization as the over-arching concept for Enterprises -- if we did this then I would expect to see an EntitlementGroup related to zero or more Entitlements, zero or more Users, and zero or more Organizations.  If we stay with the more-specific concept of an EnterpriseCustomer that is okay, but what do you think about shortening the name to just "Enterprise"?